docs/walkthrough/walkthrough-oauth2-proxy.md

```diff
@@ -0,0 +1,168 @@
+
+See supported [configuration parameters](https://oauth2-proxy.github.io/oauth2-proxy/configuration) for the full list of confiuration options.
+
+`oauth2-proxy` also supports other providers, see the list of [supported providers](https://oauth2-proxy.github.io/oauth2-proxy/auth-configuration) to setup a different provider than Github.
+
+## Setting up Dashboard Ingress rule for authentication
+
+With `oauth2-proxy` deployed, you will now secure the Tekton Dashboard `Ingress`.
+
+To do so, you need to add [annotations on the ingress](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md#external-authentication) to let `NGINX ingress controller` know that the ingress requires authentication and how to authenticate.
+
+Two annotations are used here:
+- `nginx.ingress.kubernetes.io/auth-url` indicates that the ingress uses an external service that provides authentication. This will point to `oauth2-proxy` service
+- `nginx.ingress.kubernetes.io/auth-signin` indicates the url used to sign in when the user is not logged in. This will point to `oauth2-proxy` sign in endpoint, setting the desired redirect url in the query parameters so that users are redirected to the Tekton Dashboard after logging in
+
+To apply the ingress annotations run the command below:
+
+```bash
+cat <<EOF | kubectl apply -n tekton-pipelines -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: tekton-dashboard
+  annotations:
+    nginx.ingress.kubernetes.io/auth-url: "http://oauth2-proxy.tools.svc.cluster.local/oauth2/auth"
+    nginx.ingress.kubernetes.io/auth-signin: "http://auth.127.0.0.1.nip.io/oauth2/sign_in?rd=http://\$host\$request_uri"
+spec:
+  rules:
+  - host: tekton-dashboard.127.0.0.1.nip.io
+    http:
+      paths:
+      - backend:
+          serviceName: tekton-dashboard
+          servicePort: 9097
+EOF
+```
+
+**NOTE:** The url set in the `nginx.ingress.kubernetes.io/auth-url` is the in-cluster dns name of the `oauth2-proxy` service. We can not use `auth.127.0.0.1.nip.io` here because it resolves to `127.0.0.1` and therefore the ingress controller would call itself instead of the `oauth2-proxy` service.
+
+Visiting the Tekton Dashboard url `http://tekton-dashboard.127.0.0.1.nip.io` for the first time should redirect you to the `oauth2-proxy` sign in page (`http://auth.127.0.0.1.nip.io/oauth2/sign_in?rd=http://tekton-dashboard.127.0.0.1.nip.io/`):
+
+![oauth2-proxy sign in page](./walkthrough-oauth2-proxy-sign-in.png)
+
+Clicking on the `Sign in with GitHub` button should get you on the Github OAuth sign in page:
+
+![Github OAuth sign in page](./walkthrough-oauth2-proxy-github.png)
+
+Finally, authorizing the connection should bring you back to the Tekton Dashboard.
+
+**Tip:** if you need to sign out, visit `http://auth.127.0.0.1.nip.io/oauth2/sign_out`.
```

Comment:
this should be hooked up to our LogoutButton in the Dashboard

Reply:
Yeah, if there was an easy way to it... will probably become easy once we get the installer in